packages/Adapters/POD_C/src/DTK_C_API.cpp

```diff
@@ -0,0 +1,246 @@
+    for (unsigned i = 0; i < local_num; ++i)
+    {
+      global_id = shift[comm_rank] + i;
+      for (int j = 0; j < space_dim; ++j)
+        coord[j] = points[i+local_num*j];
+      entities[i] = DataTransferKit::Point(global_id, comm_rank, coord);
+    }
+  else
+    throw std::runtime_error("Invalid data layout");
+  return entities;
+}
+
+//----------------------------------------------------------------------------//
+Teuchos::Array<double>
+  pull_field(
+    double const* data,
+    unsigned local_num,
+    DTK_Data_layout data_layout,
+    int field_dim )
+{
+  Teuchos::Array<double> field(local_num*field_dim);
+  if (data_layout == DTK_INTERLEAVE)
+    for (unsigned i = 0; i < local_num; ++i)
+      for (int j = 0; j < field_dim; ++j)
+        field[i+local_num*j] = data[field_dim*i+j];
+  else if (data_layout == DTK_CONTIGUOUS)
+    std::copy(data, data+field_dim*local_num, field.getRawPtr());
+  else
+    throw std::runtime_error("Invalid data layout");
+  return field;
+}
+
+//----------------------------------------------------------------------------//
+void push_field(
+  Teuchos::Array<double> const& field,
+  double * data,
+  DTK_Data_layout data_layout,
+  int field_dim )
+{
+  unsigned const local_num = field.size() / field_dim;
+  if (data_layout == DTK_INTERLEAVE)
+    for (unsigned i = 0; i < local_num; ++i)
+      for (int j = 0; j < field_dim; ++j)
+        data[field_dim*i+j] = field[i+local_num*j];
+  else if (data_layout == DTK_CONTIGUOUS)
+    std::copy(field.begin(), field.end(), data);
+  else
+    throw std::runtime_error("Invalid data layout");
+}
+
+//----------------------------------------------------------------------------//
+DTK_Map* DTK_Map_create( MPI_Comm        comm,
+                         double const*   src_coord,
+                         unsigned        src_num,
+                         DTK_Data_layout src_layout,
+                         double const*   tgt_coord,
+                         unsigned        tgt_num,
+                         DTK_Data_layout tgt_layout,
+                         int             space_dim,
+                         char const*     options )
+{
+  // Parse the options and build the parameter list
+  int comm_rank;
+  MPI_Comm_rank(comm, &comm_rank);
+  if (comm_rank == 0)
+    std::cout << "\nDTK_Map_create called with options:\n" << options << "\n";
+  std::stringstream ss;
+  ss.str(options);
+  boost::property_tree::ptree ptree;
+  boost::property_tree::read_info(ss, ptree);
+  // QUESTION: Should we go ahead and just parse all but the comm and ptrs?
+  Teuchos::ParameterList parameters;
+  parameters.set("Map Type", ptree.get<std::string>("Map Type"));
+  parameters.set("Spatial Dimension", ptree.get<int>("Spatial Dimension"));
+  parameters.set("Basis Type", ptree.get<std::string>("Basis Type"));
+  parameters.set("Basis Order", ptree.get<int>("Basis Order"));
+  // TODO: kNN
+  parameters.set("RBF Radius", ptree.get<double>("RBF Radius"));
```

Comment:
Did you look to see if there is a parameter list to property tree adapter class in Trilinos? I doubt there is but not sure. They have a boost::Shared_Ptr to Teuchos::RCP mapping so maybe.